dune_client/client.py

```diff
@@ -238,3 +252,125 @@ def refresh_into_dataframe(
+        they will be deleted from the query.
+        """
+        parameters: dict[str, Any] = {}
+        if name is not None:
+            parameters["name"] = name
+        if description is not None:
+            parameters["description"] = description
+        if tags is not None:
+            parameters["tags"] = tags
+        if query_sql is not None:
+            parameters["query_sql"] = query_sql
+        if params is not None:
+            parameters["parameters"] = [p.to_dict() for p in params]
+
+        if not bool(parameters):
+            # Nothing to change no need to make reqeust
+            self.logger.warning("called update_query with no proposed changes.")
+            return query_id
+
+        response_json = self._patch(
+            route=f"query/{query_id}",
+            params=parameters,
+        )
+        try:
+            # No need to make a dataclass for this since it's just a boolean.
+            return int(response_json["query_id"])
+        except KeyError as err:
+            raise DuneError(response_json, "update_query Response", err) from err
+
+    def archive_query(self, query_id: int) -> bool:
+        """
+        https://dune.com/docs/api/api-reference/edit-queries/archive-query
+        returns resulting value of Query.is_private
+        """
+        response_json = self._post(route=f"query/{query_id}/archive")
+        try:
+            # No need to make a dataclass for this since it's just a boolean.
+            return self.get_query(int(response_json["query_id"])).is_archived
+        except KeyError as err:
+            raise DuneError(response_json, "make_private Response", err) from err
+
+    def unarchive_query(self, query_id: int) -> bool:
+        """
+        https://dune.com/docs/api/api-reference/edit-queries/archive-query
+        returns resulting value of Query.is_private
+        """
+        response_json = self._post(route=f"query/{query_id}/unarchive")
+        try:
+            # No need to make a dataclass for this since it's just a boolean.
+            return self.get_query(int(response_json["query_id"])).is_archived
+        except KeyError as err:
+            raise DuneError(response_json, "make_private Response", err) from err
+
+    def make_private(self, query_id: int) -> None:
+        """
+        https://dune.com/docs/api/api-reference/edit-queries/private-query
+        returns resulting value of Query.is_private
+        """
+        response_json = self._post(route=f"query/{query_id}/private")
+        assert self.get_query(int(response_json["query_id"])).is_private
+
+    def make_public(self, query_id: int) -> None:
+        """
+        https://dune.com/docs/api/api-reference/edit-queries/private-query
+        returns resulting value of Query.is_private
+        """
+        response_json = self._post(route=f"query/{query_id}/unprivate")
+        assert not self.get_query(int(response_json["query_id"])).is_private
```

Comment:
I'm just seeing these now and from the dune reference am understanding that these are not possible to edit in the same update request. Do you think it's worth having the discussion to include those there?